pimd/pim_rp.c

```diff
@@ -1223,34 +1221,58 @@ void pim_rp_show_information(struct pim_instance *pim, struct vty *vty, bool uj)
 				strlcpy(source, "BSR", sizeof(source));
 			else
 				strlcpy(source, "None", sizeof(source));
-			if (uj) {
+			if (json) {
 				/*
 				 * If we have moved on to a new RP then add the
 				 * entry for the previous RP
 				 */
-				if (prev_rp_info
-				    && prev_rp_info->rp.rpf_addr.u.prefix4
-						       .s_addr
-					       != rp_info->rp.rpf_addr.u.prefix4
-							  .s_addr) {
+#if PIM_IPV == 4
+				if (prev_rp_info &&
+				    prev_rp_info->rp.rpf_addr.u.prefix4
+						    .s_addr !=
+					    rp_info->rp.rpf_addr.u.prefix4
+						    .s_addr) {
 					json_object_object_add(
 						json,
 						inet_ntop(AF_INET,
 							  &prev_rp_info->rp
-								  .rpf_addr.u
-								  .prefix4,
+								   .rpf_addr.u
+								   .prefix4,
+							  buf, sizeof(buf)),
+						json_rp_rows);
+					json_rp_rows = NULL;
+				}
+#else
+				if (prev_rp_info &&
+				    prev_rp_info->rp.rpf_addr.u.prefix6
+						    .s6_addr !=
+					    rp_info->rp.rpf_addr.u.prefix6
+						    .s6_addr) {
+					json_object_object_add(
+						json,
+						inet_ntop(AF_INET6,
+							  &prev_rp_info->rp
+								   .rpf_addr.u
+								   .prefix6,
 							  buf, sizeof(buf)),
 						json_rp_rows);
 					json_rp_rows = NULL;
 				}
+#endif
 
 				if (!json_rp_rows)
 					json_rp_rows = json_object_new_array();
 
 				json_row = json_object_new_object();
+#if PIM_IPV == 4
 				json_object_string_addf(
 					json_row, "rpAddress", "%pI4",
 					&rp_info->rp.rpf_addr.u.prefix4);
+#else
+				json_object_string_addf(
+					json_row, "rpAddress", "%pI6",
```

Comment:
we can use %pFX and pass &rp_info->rp.rpf_addr. It will take care of printing v4/v6.